tests/ha/control_pod/test_control_pod_soft_failures.py

```diff
@@ -0,0 +1,478 @@
+        """
+        self.ctrl_pod_list = self.node_master_list[0].get_all_pods(
+            pod_prefix=const.CONTROL_POD_NAME_PREFIX)
+        for pod in self.ctrl_pod_list:
+            self.dtm_obj.set_proc_restart_duration(self.node_master_list[0], pod,
+                                                   self.container_name, proc_restart_delay)
+
+    def get_pid(self, pod_name, container_name, proc_name):
+        """
+        Helper function to kill pid based on process name
+        :param pod_name: Name of the pod
+        :param container_name: Name of the container
+        :param proc_name: Name of the process
+        :return: tuple
+        """
+        command = cmd.GET_PID_INIT_PROCESS.format(proc_name)
+        resp = self.node_master_list[0].send_k8s_cmd(operation="exec", pod=pod_name,
+                                                     namespace=const.NAMESPACE,
+                                                     command_suffix=f"-c {container_name} "
+                                                                    f"-- {command}",
+                                                     decode=True)
+        return resp
+
+    @pytest.mark.ha
+    @pytest.mark.lc
+    @pytest.mark.tags("TEST-45499")
+    def test_iam_cruds_soft_failure_all_pods(self):
+        """
+        Test IOs and IAM user CRUDs before and after csm-agent is down in all control pods.
+        (Negative Scenario)
+        """
+        LOGGER.info("STARTED: Test IOs and IAM user CRUDs before and after csm-agent is down in"
+                    " all control pods")
+
+        proc_restart_delay = HA_CFG["common_params"]["control_proc_restart_delay"]
+        num_users = HA_CFG["s3_operation_data"]["no_csm_users"]
+        LOGGER.info("Scale replicas for control pod to %s", self.repl_num)
+        resp = self.node_master_list[0].create_pod_replicas(num_replica=self.repl_num,
+                                                            deploy=const.CONTROL_POD_NAME_PREFIX)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Total number of control pods in the cluster are: %s", self.repl_num)
+
+        LOGGER.info("Adding %s delay in config files of all control pods", proc_restart_delay)
+        self.add_proc_restart_delay(proc_restart_delay)
+        LOGGER.info("Successfully added %s delay in config files of all control pods",
+                    proc_restart_delay)
+        self.change_proc_delay = True
+
+        LOGGER.info("Step 1: Perform WRITE-READ-Verify on user %s", self.uids[0])
+        self.test_prefix = 'test-45499'
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(self.users.values())[0],
+                                                    log_prefix=self.test_prefix, skipcleanup=True)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 1: Performed WRITEs-READs-Verify with variable sizes objects.")
+
+        LOGGER.info("Create %s iam users for deletion", num_users)
```

Comment:
isn't this is done in setup_method ?

Reply:
No, in setup_method we are creating generic single user, this one is test specific.